lib/hanami/action.rb

```diff
@@ -171,16 +171,25 @@ class Action
     # @since 0.1.0
     # @api private
     def self.inherited(base)
-      base.class_eval do
-        include Utils::ClassAttribute
-        class_attribute :before_callbacks
-        self.before_callbacks = Utils::Callbacks::Chain.new
+      if base.superclass == Hanami::Action
```

Comment:
Could this be `if base.superclass = self` to eliminate the hard-coded name?

Reply:
I've got tricked by this too 😉 . In this context `self` is the superclass that you're inheriting from. For instance, if you're inheriting from `Web::Action`, then `self` is `Web::Action`.

In this case, I want only this behavior to be applied to **direct** descendants of `Hanami::Action`. By changing `Hanami::Action` with `self` causes spec failures.

Reply:
Ah yes, makes sense.